lib/internal/modules/cjs/loader.js

```diff
@@ -900,8 +900,8 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     paths = Module._resolveLookupPaths(request, parent);
   }
 
-  if (parent?.filename) {
-    if (request[0] === '#') {
+  if (request[0] === '#') {
+    if (parent?.filename) {
```

Comment:
I mean something like this:

```js
if (request[0] === '#' && (parent?.filename || parent?.id === '<repl>')) {
  const parentPath = parent?.filename ?? process.cwd() + path.sep;
```

As mentioned though I'd prefer if there's a diff that doesn't involve any change in this file, but instead ensures that the repl implementation provides enough context via `parent`.

Reply:
Got it! I duplicated the code because I wanted to make it more readable with minimal modification. Your code looks more reasonable. I'll take it.

Reply:
As to an entire REPL feature, it could not be a small change, since `cjs/loaders.js` is also shared by REPL. Am I right?